src/modules/share/share.usecase.ts

```diff
@@ -0,0 +1,165 @@
+import {
+  Injectable,
+  NotFoundException,
+  UnauthorizedException,
+} from '@nestjs/common';
+import { FileUseCases } from '../file/file.usecase';
+import { User } from '../user/user.domain';
+import { CreateShareDto } from './dto/create-share.dto';
+import { Share } from './share.domain';
+import { SequelizeShareRepository } from './share.repository';
+import crypto from 'crypto';
+import { FolderUseCases } from '../folder/folder.usecase';
+import { UpdateShareDto } from './dto/update-share.dto';
+
+@Injectable()
+export class ShareUseCases {
+  constructor(
+    private shareRepository: SequelizeShareRepository,
+    private fileUseCases: FileUseCases,
+    private folderUseCases: FolderUseCases,
+  ) {}
+
+  async getShareById(id: number) {
+    return await this.shareRepository.findById(id);
+  }
+
+  async updateShareById(
+    id: number,
+    user: User,
+    content: Partial<UpdateShareDto>,
+  ) {
+    const share = await this.shareRepository.findById(id);
+    if (share.user.id !== user.id) {
+      throw new UnauthorizedException(`You are not owner of this share`);
+    }
+    share.timesValid = content.timesValid;
+    share.active = content.active;
+
+    await this.shareRepository.update(share);
+    return share.toJSON();
+  }
+
+  async getShareByToken(token: string, user: User) {
+    const share = await this.shareRepository.findByToken(token);
+    // if is owner, not increment view
+    if ((user && !share.isOwner(user.id)) || !user) {
```

Comment:
A lot of complexity going on here. 

Try to make it clearer using boolean variables that describe what you are trying to describe on the comments, for instance:

```
const isTheOwner = user && !share.isOwner(user.id);

if (isTheOwner) {...}
```